app/helpers/application_helper.rb

```diff
@@ -184,4 +184,29 @@ def how_long_ago_label(time)
   def how_long_ago_link(url, time)
     content_tag(:a, how_long_ago_label(time), href: url)
   end
+
+  ##
+  # This replaces comment.score_for_user(user)
+  # Returns the score value, if the user can see the comment's score
+  # Returns "~", if the user can flag the comment
+  def comment_score_for_user(comment, user)
+    return {} unless user
+    return {} unless comment
+
+    score = if comment.show_score_to_user?(user)
+      comment.score
+    elsif user&.can_flag?(comment)
+      "~"
+    else
+      "&nbsp;".html_safe
+    end
+
+    score_formatted = if score.is_a?(Integer)
+      number_to_human(score, format: "%n%u")
+    else
+      score
+    end
+
+    {score_value: score, score_formatted: score_formatted}
```

Comment:
I really don't like returning an ad-hoc data structure from a function. I see that you want to have the full score on the `title` attribute and display the shortened score, but I would rather you created a second helper:

```ruby
  def upvoter_score(s)
    number_to_human(s, format: "%n%u") if s.is_a? Integer
  end
```